File: frontend/src/types/riskAnalysis.ts
// frontend/src/types/riskAnalysis.ts
/**
 * 风险分析模块类型定义
 */

// 风险等级
export type RiskLevel = 'low' | 'medium' | 'high' | 'critical';

// 分析状态
export type AnalysisStatus = 'pending' | 'parsing' | 'analyzing' | 'completed' | 'failed';

// 分析场景类型
export type SceneType = 'equity_penetration' | 'contract_risk' | 'compliance_review' | 'tax_risk';

// 文档片段引用
export interface RiskSectionRef {
  doc_id: string;
  page?: number;
  text: string;
  highlight: boolean;
}

// 风险项
export interface RiskItem {
  id: number;
  session_id: number;
  title: string;
  description: string;
  risk_level: RiskLevel;
  confidence: number;
  reasons: string[];
  suggestions: string[];
  source_type: string;
  source_rules?: number[];
  related_sections?: RiskSectionRef[];
  graph_data?: any;
  created_at: string;
}

// 风险分析会话
export interface RiskAnalysisSession {
  id: number;
  session_id: string;
  status: AnalysisStatus;
  scene_type: SceneType;
  user_description?: string;
  summary?: string;
  risk_distribution?: {
    high: number;
    medium: number;
    low: number;
  };
  total_confidence?: number;
  report_md?: string;
  created_at: string;
  completed_at?: string;
  risk_items?: RiskItem[];
}

// 提交分析请求
export interface RiskAnalysisSubmitRequest {
  scene_type: SceneType;
  user_description?: string;
  document_ids?: string[];
  enable_custom_rules?: boolean;
}

// WebSocket 进度消息
export interface WebSocketProgressMessage {
  session_id: string;
  status: AnalysisStatus;
  progress: number; // 0-1
  message: string;
}

// 节点进度消息
export interface NodeProgressMessage {
  type: 'node_progress';
  node: 'documentPreorganization' | 'multiModelAnalysis' | 'reportGeneration';
  status: 'pending' | 'processing' | 'completed' | 'failed';
  message: string;
  progress: number; // 0-1
}

// 扩展 WebSocket 消息类型
export type WebSocketMessage = WebSocketProgressMessage | NodeProgressMessage;

// 场景配置
export interface SceneConfig {
  id: SceneType;
  name: string;
  description: string;
  icon: string;
  color: string;
  placeholder: string;
}

// 场景配置预设
export const SCENE_CONFIGS: Record<SceneType, SceneConfig> = {
  equity_penetration: {
    id: 'equity_penetration',
    name: '股权穿透风险分析',
    description: '分析公司股权结构，识别潜在的控制权和关联交易风险',
    icon: 'apartment',
    color: '#722ed1',
    placeholder: '请描述需要分析的股权结构，例如：分析A公司的实际控制人和关联交易风险...'
  },
  contract_risk: {
    id: 'contract_risk',
    name: '合同条款风险分析',
    description: '识别合同条款中的法律风险点',
    icon: 'file-search',
    color: '#fa8c16',
    placeholder: '请描述合同背景和需要关注的重点...'
  },
  compliance_review: {
    id: 'compliance_review',
    name: '合规审查',
    description: '审查文档是否符合相关法律法规要求',
    icon: 'shield-check',
    color: '#52c41a',
    placeholder: '请说明合规审查的具体要求...'
  },
  tax_risk: {
    id: 'tax_risk',
    name: '税务风险分析',
    description: '识别税务相关的潜在风险',
    icon: 'calculator',
    color: '#13c2c2',
    placeholder: '请描述税务背景和关注点...'
  }
};

// API 响应类型
export interface RiskAnalysisSubmitResponse {
  session_id: string;
  status: string;
  message: string;
}

export interface RiskAnalysisUploadResponse {
  file_id: string;
  file_path: string;
  message: string;
}

export interface RiskAnalysisStartResponse {
  message: string;
  session_id: string;
}

export interface RiskAnalysisStatusResponse {
  session_id: string;
  status: string;
  summary?: string;
  risk_distribution?: {
    high: number;
    medium: number;
    low: number;
  };
}

// ==================== 风险规则包类型 ====================

// 规则项类型
export interface RiskRule {
  rule_id: string;
  rule_name: string;
  rule_prompt: string;
  priority: number;
  risk_type?: string;
  default_risk_level?: 'low' | 'medium' | 'high' | 'critical';
}

// 规则包类型
export interface RiskRulePackage {
  id: number;
  package_id: string;
  package_name: string;
  package_category: string;
  description?: string;
  applicable_scenarios?: string[];
  target_entities?: string[];
  rules: RiskRule[];
  is_active: boolean;
  is_system: boolean;
  version?: string;
  creator_id?: number;
  created_at: string;
  updated_at: string;
}

// API 响应类型 - 规则包列表
export interface RiskRulePackagesResponse {
  packages: RiskRulePackage[];
}

// 创建/更新请求类型
export interface RiskRulePackageRequest {
  package_id?: string;  // 更新时需要
  package_name: string;
  package_category: string;
  description?: string;
  applicable_scenarios?: string[];
  target_entities?: string[];
  rules: RiskRule[];
  is_active?: boolean;
}

// 规则包分类选项
export const RULE_PACKAGE_CATEGORIES = [
  { value: 'equity_risk', label: '股权风险' },
  { value: 'investment_risk', label: '投资风险' },
  { value: 'governance_risk', label: '治理风险' },
  { value: 'contract_risk', label: '合同风险' },
  { value: 'tax_risk', label: '税务风险' }
] as const;

export type RulePackageCategory = typeof RULE_PACKAGE_CATEGORIES[number]['value'];

// ==================== 评估视角类型 ====================

// 评估视角定义
export interface EvaluationPerspective {
  id: string;  // 对应规则包的 package_id
  name: string;
  category: RulePackageCategory;
  description?: string;
  priority: number;  // 1-10，数字越小优先级越高
  rule_count?: number;
  is_active?: boolean;
}

// 评估视角预设配置
export const EVALUATION_PERSPECTIVES: Omit<EvaluationPerspective, 'priority'>[] = [
  {
    id: 'company_mingling_risk',
    name: '公司混同风险',
    category: 'governance_risk',
    description: '识别人格混同、资金混同、业务混同等公司治理风险',
  },
  {
    id: 'contract_clause_risk',
    name: '合同条款风险',
    category: 'contract_risk',
    description: '分析合同条款的完整性、合法性和潜在风险',
  },
  {
    id: 'investment_project_risk',
    name: '投资项目风险',
    category: 'investment_risk',
    description: '评估投资项目的市场、财务、法律等综合风险',
  },
  {
    id: 'tax_compliance_risk',
    name: '税务风险',
    category: 'tax_risk',
    description: '识别税务合规、税收筹划、发票管理等税务风险',
  },
  {
    id: 'equity_penetration_risk',
    name: '股权穿透风险',
    category: 'equity_risk',
    description: '穿透股权结构，识别实际控制人和关联交易风险',
  },
];

// 评估视角选择器状态
export interface PerspectiveSelectorState {
  selected: EvaluationPerspective[];  // 已选择的视角（带优先级）
  available: EvaluationPerspective[];  // 可用的视角
}

File: backend/app/schemas/risk_analysis.py
# backend/app/schemas/risk_analysis.py
"""
风险评估模块的 Pydantic Schema 定义

用于 API 请求和响应的数据验证和序列化
"""

from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any
from datetime import datetime


class RiskSectionRef(BaseModel):
    """文档片段引用"""
    doc_id: str = Field(..., description="文档ID")
    page: Optional[int] = Field(None, description="页码")
    text: str = Field(..., description="原文片段")
    highlight: bool = Field(True, description="是否高亮")


class RiskItemCreate(BaseModel):
    """创建风险项请求"""
    title: str = Field(..., description="风险点标题")
    description: str = Field(..., description="详细描述")
    risk_level: str = Field(..., description="风险等级：low/medium/high/critical")
    confidence: float = Field(..., ge=0, le=1, description="置信度 0-1")
    reasons: List[str] = Field(default_factory=list, description="理由列表")
    suggestions: List[str] = Field(default_factory=list, description="规避建议列表")
    source_type: str = Field(..., description="来源类型：rule/llm/agent")
    source_rules: Optional[List[int]] = Field(None, description="来源规则 ID 列表")
    related_sections: Optional[List[RiskSectionRef]] = Field(None, description="相关文档片段")
    graph_data: Optional[Dict[str, Any]] = Field(None, description="关系图数据")


class RiskItemResponse(RiskItemCreate):
    """风险项响应"""
    id: int
    session_id: int
    created_at: datetime

    class Config:
        from_attributes = True


class RiskAnalysisSubmitRequest(BaseModel):
    """提交风险分析请求"""
    scene_type: str = Field(
        ...,
        description="分析场景：equity_penetration, contract_risk, compliance_review, tax_risk"
    )
    user_description: Optional[str] = Field(None, description="用户描述")
    document_ids: Optional[List[str]] = Field(None, description="已上传文档 ID 列表")
    enable_custom_rules: bool = Field(False, description="是否启用用户自定义规则")


class RiskAnalysisSessionResponse(BaseModel):
    """风险分析会话响应"""
    id: int
    session_id: str
    status: str
    scene_type: str
    user_description: Optional[str]
    summary: Optional[str]
    risk_distribution: Optional[Dict[str, int]]
    total_confidence: Optional[float]
    report_md: Optional[str]
    created_at: datetime
    completed_at: Optional[datetime]

    class Config:
        from_attributes = True


class RiskAnalysisDetailResponse(RiskAnalysisSessionResponse):
    """风险分析详情响应（包含风险项）"""
    risk_items: List[RiskItemResponse]
    disclaimer: str = "本评估仅供参考，不能替代专业律师意见。"


class RiskAnalysisStatusResponse(BaseModel):
    """风险分析状态响应"""
    session_id: str
    status: str
    summary: Optional[str]
    risk_distribution: Optional[Dict[str, int]]


class RiskAnalysisUploadResponse(BaseModel):
    """文档上传响应"""
    file_id: str
    file_path: str
    message: str


class RiskAnalysisStartResponse(BaseModel):
    """开始分析响应"""
    message: str
    session_id: str


# 规则管理相关 Schema
class RiskAnalysisRuleCreate(BaseModel):
    """创建风险评估规则请求"""
    name: str = Field(..., description="规则名称")
    description: Optional[str] = Field(None, description="规则描述")
    scene_type: str = Field(..., description="适用场景")
    rule_category: str = Field("custom", description="规则类型：universal/feature/custom")
    risk_type: str = Field(..., description="风险类型")
    content: str = Field(..., description="规则 Prompt 内容")
    keywords: Optional[List[str]] = Field(None, description="关键词列表")
    pattern: Optional[str] = Field(None, description="正则表达式模式")
    default_risk_level: Optional[str] = Field(None, description="默认风险等级")


class RiskAnalysisRuleResponse(BaseModel):
    """风险评估规则响应"""
    id: int
    name: str
    description: Optional[str]
    scene_type: str
    rule_category: str
    risk_type: str
    content: str
    keywords: Optional[List[str]]
    pattern: Optional[str]
    is_active: bool
    priority: int
    default_risk_level: Optional[str]
    creator_id: Optional[int]
    is_system: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
